src/Components/Web/src/Forms/FieldCssClassProvider.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Linq;
+
+namespace Microsoft.AspNetCore.Components.Forms
+{
+    /// <summary>
+    /// Supplies CSS class names for form fields to represent their validation state or other
+    /// state information from an <see cref="EditContext"/>.
+    /// </summary>
+    public class FieldCssClassProvider
```

Comment:
How do you feel about this being an abstract base type here? It makes it very clear to the user what APIs they would need to implement (in this case just one) to get the correct behavior.

Reply:
Do you propose we also have a `DefaultFieldCssClassProvider` so that people implementing a custom one still have a way of using the default behaviors in some cases?

I did consider that before but TBH it felt no better for usability. We're then relying on people discovering the `DefaultFieldCssClassProvider`.

Reply:
It's not a great deal of code to copy if they need the framework defaults. But if you think it's valuable, we can let this be.